examples/web-wagmi/src/misc/UseApproveModule.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Amount,
+  EnabledModuleFragment,
+  Erc20,
+  useApproveModule,
+  useCurrencies,
+  useEnabledModules,
+} from '@lens-protocol/react';
+import { ChangeEvent, useState } from 'react';
+
+import { Loading } from '../components/loading/Loading';
+
+type UseApproveModuleFormProps = {
+  currencies: Erc20[];
+  modules: EnabledModuleFragment[];
+};
+
+function UseApproveModuleForm({ currencies, modules }: UseApproveModuleFormProps) {
+  const [amount, setAmount] = useState(0);
+  const [selectedCurrency, setSelectedCurrency] = useState(0);
+  const [selectedModule, setSelectedModule] = useState(0);
+
+  const { approve, isPending, isApproved } = useApproveModule({
+    spender: modules[selectedModule].contractAddress,
+    amount: Amount.erc20(currencies[selectedCurrency], amount),
+  });
+
+  const handleAmountChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.value) setAmount(0);
+    else setAmount(e.target.valueAsNumber);
+  };
+
+  const handleCurrencyChange = (e: ChangeEvent<HTMLSelectElement>) =>
+    setSelectedCurrency(Number(e.target.value));
+
+  const handleModuleChange = (e: ChangeEvent<HTMLSelectElement>) =>
+    setSelectedModule(Number(e.target.value));
```

Comment:
seems not working as you don't set `value` as number so it's using `moduleName` which is a string so it always set's the `Nan` as `selectedModule`